pkg/build/generator/generator.go

```diff
@@ -356,6 +392,17 @@ func (g *BuildGenerator) Clone(ctx kapi.Context, request *buildapi.BuildRequest)
 	// Copy build trigger information to the build object.
 	newBuild.Spec.TriggeredBy = request.TriggeredBy
 
+	// Update the Docker build args
+	if request.StrategyOptions != nil && request.StrategyOptions.DockerOptions != nil {
+		dockerOpts := request.StrategyOptions.DockerOptions
+		if dockerOpts.BuildArgs != nil && len(dockerOpts.BuildArgs) > 0 {
+			if newBuild.Spec.Strategy.DockerStrategy == nil {
+				return nil, errors.NewBadRequest(fmt.Sprintf("Cannot specify build args on %s/%s, not a Docker build.", buildConfig.Namespace, buildConfig.ObjectMeta.Name))
+			}
+			newBuild.Spec.Strategy.DockerStrategy.BuildArgs = updateBuildArgs(&newBuild.Spec.Strategy.DockerStrategy.BuildArgs, dockerOpts.BuildArgs)
+		}
+	}
```

Comment:
is the clone path ignoring the request.environment field?  it looks like it is.  that should be fixed while you're in here.